kayenta-objectstore-configbin/src/main/java/com/netflix/kayenta/configbin/storage/ConfigBinStorageService.java

```diff
@@ -285,27 +285,49 @@ public void deleteObject(String accountName, ObjectType objectType, String objec
       String ownerApp = credentials.getOwnerApp();
       String configType = credentials.getConfigType();
       ConfigBinRemoteService remoteService = credentials.getRemoteService();
-      String jsonBody =
-          AuthenticatedRequest.allowAnonymous(
-              () ->
-                  retry.retry(
-                      () -> remoteService.list(ownerApp, configType), MAX_RETRIES, RETRY_BACKOFF));
-
-      try {
-        List<String> ids =
-            kayentaObjectMapper.readValue(jsonBody, new TypeReference<List<String>>() {});
-
-        if (ids.size() > 0) {
-          return ids.stream().map(i -> metadataFor(credentials, i)).collect(Collectors.toList());
+      List<String> ids = new ArrayList<>();
+      boolean hasNext = true;
+      String pageId = null;
+      while (hasNext) {
+        try {
+          final String currentPage = pageId;
+          String jsonBody =
+              AuthenticatedRequest.allowAnonymous(
+                  () ->
+                      retry.retry(
+                          () -> remoteService.list(ownerApp, configType, currentPage),
+                          MAX_RETRIES,
+                          RETRY_BACKOFF));
+          ConfigBinListResponse response =
+              kayentaObjectMapper.readValue(jsonBody, ConfigBinListResponse.class);
+          hasNext = response.hasNext;
+          pageId = response.nextPageId;
+          ids.addAll(
+              response.nameVersions.stream().map(nv -> nv.configName).collect(Collectors.toList()));
+        } catch (IOException e) {
+          log.error("List failed on path {}: {}", ownerApp, e);
+          return Collections.emptyList();
         }
-      } catch (IOException e) {
-        log.error("List failed on path {}: {}", ownerApp, e);
+      }
+
+      if (ids.size() > 0) {
```

Comment:
NIT - I don't think you actually need this check...
If you stream an empty list of IDs, then you would just end up collecting an empty list. Doesn't hurt anything, but may simplify the branching.

Reply:
Nice catch. This is a hold over from the old code. I sometimes get blinders on when I'm doing something a bit more surgical.